diff -ur globus_ftp_client-3.14.orig/globus_ftp_client_restart_marker.c globus_ftp_client-3.14/globus_ftp_client_restart_marker.c
--- globus_ftp_client-3.14.orig/globus_ftp_client_restart_marker.c	2008-10-02 05:46:48.000000000 +0200
+++ globus_ftp_client-3.14/globus_ftp_client_restart_marker.c	2009-06-04 05:32:21.000000000 +0200
@@ -666,7 +666,7 @@
 {
     globus_off_t				offset;
     globus_off_t				end;
-    globus_size_t				consumed;
+    int						consumed;
     globus_object_t *				err;
     globus_result_t				res;
     const char *				p;
