diff -ur globus_xio-2.7.orig/pkgdata/pkg_data_src.gpt.in globus_xio-2.7/pkgdata/pkg_data_src.gpt.in
--- globus_xio-2.7.orig/pkgdata/pkg_data_src.gpt.in	2008-04-30 20:25:28.000000000 +0200
+++ globus_xio-2.7/pkgdata/pkg_data_src.gpt.in	2008-07-14 22:53:51.000000000 +0200
@@ -3,7 +3,7 @@
 
 <gpt_package_metadata Format_Version="0.02" Name="globus_xio" >
 
-  <Aging_Version Age="5" Major="2" Minor="8" />
+  <Aging_Version Age="2" Major="2" Minor="8" />
   <Description >Globus XIO Framework</Description>
   <Functional_Group >Communication</Functional_Group>
   <Version_Stability Release="Experimental" Dirt="@DIRT_TIMESTAMP@-@DIRT_BRANCH_ID@" />
