diff -ur globus_proxy_utils-3.7.orig/programs/grid_proxy_init.c globus_proxy_utils-3.7/programs/grid_proxy_init.c
--- globus_proxy_utils-3.7.orig/programs/grid_proxy_init.c	2008-09-09 15:55:12.000000000 +0200
+++ globus_proxy_utils-3.7/programs/grid_proxy_init.c	2010-05-15 19:57:29.935498747 +0200
@@ -1172,8 +1172,8 @@
     int                                 pci_NID;
     int                                 pci_old_NID;
 
-    pci_NID = OBJ_sn2nid(PROXYCERTINFO_SN);
-    pci_old_NID = OBJ_sn2nid(PROXYCERTINFO_OLD_SN);
+    pci_NID = OBJ_txt2nid(PROXYCERTINFO_OID);
+    pci_old_NID = OBJ_txt2nid(PROXYCERTINFO_OLD_OID);
     extension_object = X509_EXTENSION_get_object(extension);
     nid = OBJ_obj2nid(extension_object);
 
