diff -ur globus_core-5.15.orig/doxygen/globus-build-doxygen-dependencies globus_core-5.15/doxygen/globus-build-doxygen-dependencies
--- globus_core-5.15.orig/doxygen/globus-build-doxygen-dependencies	2008-10-21 09:57:03.000000000 +0200
+++ globus_core-5.15/doxygen/globus-build-doxygen-dependencies	2008-10-21 10:09:07.000000000 +0200
@@ -77,7 +77,14 @@
 pod2usage('Must specify src_metadata_file') if !defined($srcfile);
 
 @deps = process_package($srcfile);
-print join(' ', map {"$prefix/share/doc/$_/html/$_.tag"} @deps) . "\n";
+foreach (@deps) {
+    my $pkg = new Grid::GPT::V1::Package;
+    $pkg->read_metadata_file("$prefix/share/globus/packages/$_/pkg_data_noflavor_doc.gpt");
+    my $dir = "$_-$pkg->{'Version'}->{'major'}.$pkg->{'Version'}->{'minor'}";
+    $dir =~ s/_/-/g;
+    $_ = "$prefix/share/doc/$dir/html/$_.tag";
+}
+print join(' ', @deps) . "\n";
 
 sub process_package($)
 {
