diff -ur globus_rls_client-5.2.orig/library/rpc.c globus_rls_client-5.2/library/rpc.c
--- globus_rls_client-5.2.orig/library/rpc.c	2009-10-22 02:00:26.000000000 +0200
+++ globus_rls_client-5.2/library/rpc.c	2010-01-27 11:05:58.038905559 +0100
@@ -392,7 +392,8 @@
   if (rc != GLOBUS_RLS_SUCCESS)
     return rc;
   if (nb != b->len) {
-    sprintf(errmsg, "Attempted to write %lu bytes, only wrote %lu", b->len, nb);
+    sprintf(errmsg, "Attempted to write %lu bytes, only wrote %lu",
+	    (unsigned long)b->len, (unsigned long)nb);
     return GLOBUS_RLS_GLOBUSERR;
   }
   b->len = 0;
