diff -ur globus_gsi_proxy_ssl-2.2.orig/library/proxypolicy.c globus_gsi_proxy_ssl-2.2/library/proxypolicy.c
--- globus_gsi_proxy_ssl-2.2.orig/library/proxypolicy.c	2010-02-19 22:18:52.000000000 +0100
+++ globus_gsi_proxy_ssl-2.2/library/proxypolicy.c	2010-05-31 11:40:34.468367904 +0200
@@ -72,7 +72,7 @@
     ret = NULL;
 
     M_ASN1_New_Malloc(ret, PROXYPOLICY);
-    ret->policy_language = OBJ_nid2obj(OBJ_sn2nid(IMPERSONATION_PROXY_SN));
+    ret->policy_language = OBJ_nid2obj(OBJ_txt2nid(IMPERSONATION_PROXY_OID));
     ret->policy = NULL;
     return (ret);
     M_ASN1_New_Error(ASN1_F_PROXYPOLICY_NEW);
