diff -ur globus_xio_gsi_driver-0.6.orig/globus_xio_gsi.c globus_xio_gsi_driver-0.6/globus_xio_gsi.c
--- globus_xio_gsi_driver-0.6.orig/globus_xio_gsi.c	2008-09-18 22:24:06.000000000 +0200
+++ globus_xio_gsi_driver-0.6/globus_xio_gsi.c	2008-12-31 13:45:31.000000000 +0100
@@ -577,11 +577,11 @@
     
     if(driver_server)
     {
-        result = globus_l_xio_gsi_attr_copy((void **) &attr, driver_server);
+        result = globus_l_xio_gsi_attr_copy((void **) (void *) &attr, driver_server);
     }
     else
     {
-        result = globus_l_xio_gsi_attr_init((void **) &attr);
+        result = globus_l_xio_gsi_attr_init((void **) (void *) &attr);
     }
     if(result != GLOBUS_SUCCESS)
     {
@@ -1715,7 +1715,7 @@
     
     if(driver_attr)
     {
-        result = globus_l_xio_gsi_attr_copy((void **) &handle->attr,
+        result = globus_l_xio_gsi_attr_copy((void **) (void *) &handle->attr,
                                             driver_attr);
         if(result == GLOBUS_SUCCESS && driver_link)
         {
@@ -1724,13 +1724,13 @@
     }
     else if(driver_link)
     {
-        result = globus_l_xio_gsi_attr_copy((void **) &handle->attr,
+        result = globus_l_xio_gsi_attr_copy((void **) (void *) &handle->attr,
                                             driver_link);
     }
     else
     {
         result = globus_l_xio_gsi_attr_copy(
-            (void **) &handle->attr,
+            (void **) (void *) &handle->attr,
             (void *) &globus_l_xio_gsi_attr_default);        
     }
     
