diff -ur gpt-3.2.orig/packaging_tools/etc/globus_package.dtd gpt-3.2/packaging_tools/etc/globus_package.dtd
--- gpt-3.2.orig/packaging_tools/etc/globus_package.dtd	2004-05-20 19:42:17.000000000 +0200
+++ gpt-3.2/packaging_tools/etc/globus_package.dtd	2008-07-02 10:07:19.000000000 +0200
@@ -3,8 +3,8 @@
                                 Description, 
                                 Functional_Group, 
                                 Version_Stability, 
-                                VersionLabel, 
-                                PackagingTool, 
+                                VersionLabel?, 
+                                PackagingTool?, 
                                 (data_pkg       | 
                                  dev_pkg        | 
                                  doc_pkg        | 
@@ -46,41 +46,43 @@
 	ToolName CDATA #REQUIRED
 	ToolVersion CDATA #REQUIRED
 >
-<!ELEMENT src_pkg (Version_Label,
+<!ELEMENT src_pkg (Version_Label?,
                    With_Flavors, 
                    Source_Dependencies*, 
                    Source_Setup_Dependency*, 
-                   Build_Instructions, 
-                   Environment, 
-                   Post_Install_Message, 
-                   Post_Install_Program, 
-                   Setup)>
-<!ELEMENT pgm_pkg (Version_Label,
+                   Build_Instructions?, 
+                   Build_Environment, 
+                   Post_Install_Message*, 
+                   Post_Install_Program*, 
+                   Setup*)>
+<!ELEMENT pgm_pkg (Version_Label?,
                    Flavor, 
                    Binary_Dependencies*, 
                    Setup_Dependency*, 
                    Post_Install_Message*, 
                    Post_Install_Program*, 
                    Setup*)>
-<!ELEMENT pgm_static_pkg (Version_Label,
+<!ELEMENT pgm_static_pkg (Version_Label?,
                           Flavor, 
                           Binary_Dependencies*, 
                           Setup_Dependency*, 
                           Post_Install_Message*, 
                           Post_Install_Program*, 
                           Setup*)>
-<!ELEMENT dev_pkg (Version_Label,
+<!ELEMENT dev_pkg (Version_Label?,
                    Flavor, 
                    Binary_Dependencies*, 
-                   Environment?)>
-<!ELEMENT rtl_pkg (Version_Label,
+                   Build_Environment?)>
+<!ELEMENT rtl_pkg (Version_Label?,
                    Flavor, 
                    Binary_Dependencies*, 
                    Setup_Dependency*)>
-<!ELEMENT data_pkg (Version_Label,
+<!ELEMENT data_pkg (Version_Label?,
+                    Flavor?, 
                     Binary_Dependencies*, 
                     Setup_Dependency*)>
-<!ELEMENT doc_pkg (Version_Label,
+<!ELEMENT doc_pkg (Version_Label?,
+                   Flavor?, 
                    Binary_Dependencies*, 
                    Setup_Dependency*)>
 <!ELEMENT Flavor (#PCDATA)>
@@ -145,28 +147,22 @@
                  pgm  | 
                  rtl) #REQUIRED
 >
-<!ELEMENT Environment (Cflags?, 
-                             PackageLibs?, 
-                             ExternalLibs?, 
-                             ExternalIncludes?
-                             Variable?
-                             SourcedFile?
+<!ELEMENT Build_Environment (cflags?, 
+                             external_includes?, 
+                             pkg_libs?, 
+                             external_libs?
                       )>
-<!ELEMENT Cflags (#PCDATA)>
-<!ELEMENT ExternalIncludes (#PCDATA)>
-<!ELEMENT PackageLibs (#PCDATA)>
-<!ELEMENT ExternalLibs (#PCDATA)>
-<!ELEMENT Variable EMPTY>
-<!ATTLIST Variable Name CDATA #REQUIRED Value CDATA #REQUIRED>
-<!ELEMENT SourcedFile (#PCDATA)>
-<!ATTLIST SourcedFile Shell (sh | csh) #REQUIRED>
+<!ELEMENT cflags (#PCDATA)>
+<!ELEMENT external_includes (#PCDATA)>
+<!ELEMENT pkg_libs (#PCDATA)>
+<!ELEMENT external_libs (#PCDATA)>
 
 <!ELEMENT Build_Instructions (Build_Step*, flavor_choice*)>
-<!ATTLIST Build_Instructions SrcDir CDATA>
+<!ATTLIST Build_Instructions SrcDir CDATA #IMPLIED>
 
 <!ELEMENT Build_Step (#PCDATA)>
 <!ATTLIST Build_Step
-	Macro_Args CDATA #REQUIRED
+	Macro_Args CDATA #IMPLIED
 >
 <!ELEMENT Setup (Aging_Version)>
 <!ATTLIST Setup
