diff -ur globus_gsi_credential-3.3.orig/library/globus_gsi_credential.c globus_gsi_credential-3.3/library/globus_gsi_credential.c
--- globus_gsi_credential-3.3.orig/library/globus_gsi_credential.c	2008-11-12 18:13:03.000000000 +0100
+++ globus_gsi_credential-3.3/library/globus_gsi_credential.c	2010-05-15 20:30:45.423249163 +0200
@@ -1862,8 +1862,8 @@
 
     *proxycertinfo = NULL;
 
-    pci_NID = OBJ_sn2nid(PROXYCERTINFO_SN);
-    pci_old_NID = OBJ_sn2nid(PROXYCERTINFO_OLD_SN);
+    pci_NID = OBJ_txt2nid(PROXYCERTINFO_OID);
+    pci_old_NID = OBJ_txt2nid(PROXYCERTINFO_OLD_OID);
     if(pci_NID == NID_undef || pci_old_NID == NID_undef)
     {
         GLOBUS_GSI_CRED_ERROR_RESULT(
