diff -ur globus_xio_gsi_driver-0.6.orig/configure.in globus_xio_gsi_driver-0.6/configure.in
--- globus_xio_gsi_driver-0.6.orig/configure.in	2006-04-21 21:10:34.000000000 +0200
+++ globus_xio_gsi_driver-0.6/configure.in	2008-10-21 07:12:14.000000000 +0200
@@ -12,7 +12,7 @@
 dnl Initialize the automake rules the last argument
 AM_INIT_AUTOMAKE($GPT_NAME, $GPT_VERSION)
 
-LAC_DOXYGEN("../", "", "$GLOBUS_LOCATION/sbin/globus-vararg-enums-doxygen-filter.pl")
+LAC_DOXYGEN("../", "", "${GLOBUS_LOCATION:-/usr}/sbin/globus-vararg-enums-doxygen-filter.pl")
 
 GLOBUS_FINALIZE
 
