diff -ur globus_rls_client-5.2.orig/Doc/man/Makefile.am globus_rls_client-5.2/Doc/man/Makefile.am
--- globus_rls_client-5.2.orig/Doc/man/Makefile.am	2002-12-12 02:47:06.000000000 +0100
+++ globus_rls_client-5.2/Doc/man/Makefile.am	2010-01-23 11:45:07.657532753 +0100
@@ -1,7 +1,6 @@
 include $(top_srcdir)/globus_automake_pre
 
 SUBDIRS= \
-	man1 \
-	man3
+	man1
 
 include $(top_srcdir)/globus_automake_post
diff -ur globus_rls_client-5.2.orig/library/globus_rls_client.h globus_rls_client-5.2/library/globus_rls_client.h
--- globus_rls_client-5.2.orig/library/globus_rls_client.h	2009-10-26 18:10:27.000000000 +0100
+++ globus_rls_client-5.2/library/globus_rls_client.h	2010-01-23 11:45:07.658533647 +0100
@@ -1122,7 +1122,7 @@
  *	all results.
  *
  * @param str2_list
- *	List of LFNs that map to @pfn.  Each list datum will be a
+ *	List of LFNs that map to @a pfn.  Each list datum will be a
  *	globus_rls_string2_t structure.  <B>s1</B> will
  *	be the LFN, and <B>s2</B> the PFN it maps to.
  *
@@ -1186,7 +1186,7 @@
  *	all results.
  *
  * @param str2_list
- *	List of LFNs that map to @pfn_pattern.  Each list datum will be a
+ *	List of LFNs that map to @a pfn_pattern.  Each list datum will be a
  *	globus_rls_string2_t structure.  <B>s1</B> will
  *	be the LFN, and <B>s2</B> the PFN it maps to.
  *
@@ -1223,7 +1223,7 @@
  *	all results.
  *
  * @param str2_list
- *	List of PFNs that map to @lfn.  Each list datum will be a
+ *	List of PFNs that map to @a lfn.  Each list datum will be a
  *	globus_rls_string2_t structure.  <B>s1</B> will
  *	be the LFN, and <B>s2</B> the PFN it maps to.
  *
@@ -1285,7 +1285,7 @@
  *	all results.
  *
  * @param str2_list
- *	List of PFNs that map to @lfn_pattern.  Each list datum will be a
+ *	List of PFNs that map to @a lfn_pattern.  Each list datum will be a
  *	globus_rls_string2_t structure.  <B>s1</B> will
  *	be the LFN, and <B>s2</B> the PFN it maps to.
  *
@@ -1486,8 +1486,8 @@
  *
  * @param str2_list
  *	Results added to list.  Datums in @a str2_list are of type
- *	globus_rls_string2_t structure.  <B>s1<B> will be the rli url,
- *	<B>s2<B> an  empty string or the pattern used to partition
+ *	globus_rls_string2_t structure.  <B>s1</B> will be the rli url,
+ *	<B>s2</B> an  empty string or the pattern used to partition
  *	updates.  See @ref globus_rls_client_queryresult.
  *
  * @retval GLOBUS_SUCCESS
@@ -1646,7 +1646,7 @@
  *	all results.
  *
  * @param str2_list
- *	List of LRCs that @lfn maps to.  Each list datum will be a
+ *	List of LRCs that @a lfn maps to.  Each list datum will be a
  *	globus_rls_string2_t structure.  <B>s1</B> will
  *	be the LFN, and <B>s2</B> the LRC it maps to.  @a str2_list
  *	should be freed with globus_rls_client_free_list().
@@ -1710,7 +1710,7 @@
  *	all results.
  *
  * @param str2_list
- *	List of LRCs that map to @lfn_pattern.  Each list datum will be a
+ *	List of LRCs that map to @a lfn_pattern.  Each list datum will be a
  *	globus_rls_string2_t.  <B>s1</B> will
  *	be the LFN, and <B>s2</B> the LRC it maps to.  @a str2_list
  *	should be freed with globus_rls_client_free_list().
@@ -1851,8 +1851,8 @@
  *
  * @param str2_list
  *	Results added to list.  Datums in @a str2_list are of type
- *	globus_rls_string2_t structure.  <B>s1<B> will be the rli url,
- *	<B>s2<B> an  empty string or the pattern used to partition
+ *	globus_rls_string2_t structure.  <B>s1</B> will be the rli url,
+ *	<B>s2</B> an  empty string or the pattern used to partition
  *	updates.  See @ref globus_rls_client_queryresult.
  *
  * @retval GLOBUS_SUCCESS
@@ -2098,7 +2098,7 @@
  *
  * @param r
  *	Result returned by RLS API function.  @a r is freed by this call
- *	and should not be referenced again.  If @preserve is set then a
+ *	and should not be referenced again.  If @a preserve is set then a
  *	new globus_result_t is constructed with the same values and
  *	returned as the function value.
  *
