diff -ur globus_rls_client-5.1.orig/cli.c globus_rls_client-5.1/cli.c
--- globus_rls_client-5.1.orig/cli.c	2008-05-30 18:33:29.000000000 +0200
+++ globus_rls_client-5.1/cli.c	2009-05-09 11:19:17.000000000 +0200
@@ -295,7 +295,8 @@
   char			buf[BUFSIZ];
   int			iargc;
   char			*iargv[MAXCMD];
-  struct        sigaction	sa;
+#ifndef WIN32
+  struct sigaction	sa;
 
   sigemptyset(&sa.sa_mask);
   sa.sa_handler = hsig;
@@ -303,6 +304,7 @@
 
   sigaction(SIGINT, &sa, NULL);
   sigaction(SIGTERM, &sa, NULL); 
+#endif
 
   for (i = 0; i < NMODS; i++)
     if ((rc = globus_module_activate(modules[i])) != GLOBUS_SUCCESS) {
diff -ur globus_rls_client-5.1.orig/library/client.c globus_rls_client-5.1/library/client.c
--- globus_rls_client-5.1.orig/library/client.c	2006-06-01 02:37:03.000000000 +0200
+++ globus_rls_client-5.1/library/client.c	2009-05-09 11:02:29.000000000 +0200
@@ -1593,7 +1593,7 @@
 
   switch (type) {
     case globus_rls_attr_type_date:
-      if (!strptime(sval, "%Y-%m-%d %H:%M:%S", &t))
+      if (!globus_strptime(sval, "%Y-%m-%d %H:%M:%S", &t))
         return mkresult(GLOBUS_RLS_BADARG, sval);
       t.tm_isdst = -1;
       attr->val.t = mktime(&t);
diff -ur globus_rls_client-5.1.orig/library/rpc.c globus_rls_client-5.1/library/rpc.c
--- globus_rls_client-5.1.orig/library/rpc.c	2006-06-01 02:37:03.000000000 +0200
+++ globus_rls_client-5.1/library/rpc.c	2009-05-09 11:02:29.000000000 +0200
@@ -126,7 +126,7 @@
 {
   globus_result_t   r;
   IOMON             mon;
-  struct timespec   ts;
+  globus_abstime_t  ts;
   globus_bool_t     timed_out = GLOBUS_FALSE;
 
   globus_mutex_init(&mon.mtx, GLOBUS_NULL);
@@ -183,7 +183,7 @@
   globus_result_t	r;
   globus_size_t		try_nb;
   IOMON			mon;
-  struct timespec	ts;
+  globus_abstime_t	ts;
 
   r = globus_io_try_read(h, buf, max_nb, nbr);
   if (r != GLOBUS_SUCCESS)
@@ -252,7 +252,7 @@
   globus_result_t	r;
   globus_size_t		try_nb;
   IOMON			mon;
-  struct timespec	ts;
+  globus_abstime_t	ts;
 
   r = globus_io_try_write(h, buf, nb, nbw);
   if (r != GLOBUS_SUCCESS)
@@ -305,7 +305,7 @@
 {
   globus_result_t	r;
   IOMON			mon;
-  struct timespec	ts;
+  globus_abstime_t	ts;
 
   globus_mutex_init(&mon.mtx, GLOBUS_NULL);
   globus_cond_init(&mon.cond, GLOBUS_NULL);
